### PR TITLE
Add direct arg to login flow

### PR DESCRIPTION
**Pull request by CarlSchwan**
Signed-off-by: Vincent Petry <vincent@nextcloud.com>
Co-Authored-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1: see minor comment for consistency

File: core/templates/loginflow/authpicker.php [hunk 1]
```diff
@@ -62,6 +62,7 @@
 		</p>
 		<input type="hidden" name="stateToken" value="<?php p($_['stateToken']) ?>" />
 		<input type="hidden" name="requesttoken" value="<?php p($_['requesttoken']) ?>">
+		<input type="hidden" name="direct" value="<?php p($_['direct']) ?>">
```
[the remark this answers is not in the document]
Comment (CarlSchwan) [reply core/templates/loginflow/authpicker.php:65]: done

**Review by come-nc — APPROVED** [review]

**Comment by CarlSchwan** [comment]
Tests are failing

File: core/templates/loginflow/grant.php [hunk 2]
```diff
@@ -39,14 +39,17 @@
...
-			<input type="hidden" name="clientIdentifier" value="<?php p($_['clientIdentifier']) ?>" />	
-			<input type="hidden" name="requesttoken" value="<?php p($_['requesttoken']) ?>" />	
-			<input type="hidden" name="stateToken" value="<?php p($_['stateToken']) ?>" />	
-			<input type="hidden" name="oauthState" value="<?php p($_['oauthState']) ?>" />	
+		<form method="POST" action="<?php p($urlGenerator->linkToRouteAbsolute('core.ClientFlowLogin.generateAppPassword')) ?>">
+			<input type="hidden" name="clientIdentifier" value="<?php p($_['clientIdentifier']) ?>" />
+			<input type="hidden" name="requesttoken" value="<?php p($_['requesttoken']) ?>" />
+			<input type="hidden" name="stateToken" value="<?php p($_['stateToken']) ?>" />
+			<input type="hidden" name="oauthState" value="<?php p($_['oauthState']) ?>" />
+			<?php if (p($_['direct'])) { ?>
```
Comment (skjnldsv) [inline core/templates/loginflow/grant.php:47]: Helloooo :D

Comment (CarlSchwan) [reply core/templates/loginflow/grant.php:47]: outch

Comment (CarlSchwan) [reply core/templates/loginflow/grant.php:47]: https://github.com/nextcloud/server/pull/32374

**Pull request merged**